src/renderer/base/renderer.cpp

```diff
@@ -123,6 +123,9 @@ Renderer::~Renderer()
     // Force scope exit end paint to finish up collecting information and possibly painting
     endPaint.reset();
 
+    // 7. Paint terminal effects
+    RETURN_IF_FAILED(_PaintTerminalEffects(pEngine));

```

Comment:
This has to come after `end paint`? Can you explain why?

If so, maybe it should just be an optional thing that happens in `DxEngine::EndPaint` instead of introducing a new interface definition that only applies to `DxEngine`...

Reply:
Ah good idea